x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processor_form/processors/script.tsx

```diff
@@ -122,6 +129,17 @@ const fieldsConfig: FieldsConfig = {
 
 export const Script: FormFieldsComponent = ({ initialFieldValues }) => {
   const [showId, setShowId] = useState(() => !!initialFieldValues?.id);
+  const [scriptLanguage, setScriptLanguage] = useState<string>('plaintext');
+
+  const [{ fields }] = useFormData();
```

Comment:
```suggestion
  const [{ fields }] = useFormData({ watch: 'fields.lang' });
```

This will make sure we don't re-render unless needed otherwise there is some sluggishness on text inputs.